pkg/utils/getpartitions.go

```diff
@@ -53,13 +53,45 @@ func GetAllPartitions(runner v1.Runner) (v1.PartitionList, error) {
 
 	// --list : show each partition only once
 	// --bytes: don't show sizes in human readable format but rather number of bytes
-	out, err := runner.Run("lsblk", "--list", "--bytes",
-		"/dev/disk/by-path/*", "-o", "NAME,PKNAME,PATH,FSTYPE,MOUNTPOINT,SIZE,RO,LABEL", "-J")
+	devices, err := filepath.Glob("/dev/disk/by-path/*")
 	if err != nil {
-		return parts, fmt.Errorf("lsblk failed with: %w\nOutput: %s", err, string(out))
+		return parts, fmt.Errorf("lsblk failed with: %w", err)
+	}
+	for _, dev := range devices {
+		out, err := runner.Run("lsblk", "--list", "--bytes",
+			dev, "-o", "NAME,PKNAME,PATH,FSTYPE,MOUNTPOINT,SIZE,RO,LABEL", "-J")
+		if err != nil {
+			return parts, fmt.Errorf("lsblk failed with: %w\nOutput: %s", err, string(out))
+		}
+		p := PartitionFromLsblk(string(out))
+		if p != nil {
+			parts = append(parts, p)
+		}
+
+	}
+
+	return parts, err
+}
+
+func PartitionFromLsblk(lsblkOutput string) *v1.Partition {
+	var part = &v1.Partition{}
+	lsblk := &Lsblk{}
+	err := json.Unmarshal([]byte(lsblkOutput), lsblk)
+	if err != nil {
+		return part
+	}
+	if len(lsblk.BlockDevices) == 1 {
+		b := lsblk.BlockDevices[0]
+		part.Name = b.Name
```

Comment:
Im missing the parent stuff, is that important? I could not understand why its needed

Reply:
There is some place in the code where we use the value of `Disk` or one of the partitions in order to do the "reset" iirc. I need to look it up. Since I didn't know in how many places we rely on the `Disk` to be set correctly (either now or in the future), I tried to populate it with the correct value.

Reply:
makes sense, added!